express/scripts/scripts.js

```diff
@@ -1065,78 +1076,250 @@ export function checkTesting() {
+ * like "Audience", "Status" or "Blocks".
+ *
+ * "experience" hosts the experiences in columns, consisting of:
+ * a "Percentage Split", "Label" and a set of "Pages".
+ *
+ *
+ * @param {string} experimentid
+ * @returns {object} containing the experiment manifest
+ */
+
+export async function fetchExperimentConfig(experiment) {
+  const path = `/express/experiments/${experiment}/manifest.json`;
+  try {
+    const config = {};
+    const resp = await fetch(path);
+    const json = await resp.json();
+    json.settings.data.forEach((line) => {
+      const key = toCamelCase(line.Name);
+      let value = line.Value;
+      if (key === 'blocks') value = value.split(',').map((e) => e.trim().toLowerCase());
+      config[key] = value;
+    });
+    config.id = experiment;
+
+    const variants = {};
+    let variantNames = Object.keys(json.experiences.data[0]);
+    variantNames.shift();
+    variantNames = variantNames.map((vn) => toCamelCase(vn));
+    variantNames.forEach((variantName) => {
+      variants[variantName] = {};
     });
 
-    const testSetup = [];
-
-    if ($testTable) {
-      $testTable.querySelectorAll('tr').forEach(($row) => {
-        const $name = $row.children[0];
-        const $percentage = $row.children[1];
-        const $a = $name.querySelector('a');
-        if ($a) {
-          const url = new URL($a.href);
-          testSetup.push({
-            url: url.pathname,
-            traffic: parseFloat($percentage.textContent) / 100.0,
-          });
+    let lastKey = 'default';
+
+    json.experiences.data.forEach((line) => {
+      let key = toCamelCase(line.Name);
+      if (!key) key = lastKey;
+      lastKey = key;
+      const vns = Object.keys(line);
+      vns.shift();
+      vns.forEach((vn) => {
+        const camelVN = toCamelCase(vn);
+        if (key === 'pages') {
+          variants[camelVN][key] = variants[camelVN][key] || [];
+          variants[camelVN][key].push(new URL(line[vn]).pathname);
+        } else {
+          variants[camelVN][key] = line[vn];
         }
       });
+    });
+    config.variants = variants;
+    config.variantNames = variantNames;
+    console.log(config);
+    return config;
+  } catch (e) {
+    console.log(`error loading experiment manifest: ${path}`, e);
+  }
+  return null;
+}
+
+/**
+ * Replaces main with content from path
+ * @param {string} path
+ */
+
+async function replaceInner(path, element) {
+  const plainPath = `${path}.plain.html`;
+  try {
+    const resp = await fetch(plainPath);
+    const html = await resp.text();
+    element.innerHTML = html;
+  } catch (e) {
+    console.log(`error loading experiment content: ${plainPath}`, e);
+  }
+  return null;
+}
+
+/**
+ * this is an extensible stub to take on audience mappings
+ * @param {string} audience
+ * @return {boolean} is member of this audience
+ */
+
+function checkExperimentAudience(audience) {
+  if (audience === 'mobile') {
+    return window.innerWidth < 600;
```

Comment:
I know you are opposed to using `navigator.userAgent` to identify mobile devices, but I think solely relying on the window width can give ambiguous results. For example, at this breakpoint, phones in landscape and tablets in any orientation will be treated as desktop.
[CCX0033](https://github.com/adobe/express-website/pull/423) is an example where the audience "mobile" definitely includes phones and tablets, and desktops must be excluded, regardless of window width.